core/kazoo_number_manager/src/carriers/knm_carriers.erl

```diff
@@ -53,145 +53,149 @@ find(Num, Quantity) ->
-                                         kz_json:objects().
-check_existing_phone_number(Number, Acc, PhoneNumber) ->
-    case lists:member(knm_phone_number:state(PhoneNumber), ?KNM_AVAILABLE_STATES) of
-        'true' -> [found_number_to_jobj(Number) | Acc];
-        'false' -> Acc
+    case knm_number:save(knm_number:set_phone_number(Number, PhoneNumber)) of
+        {'ok', SavedNumber} ->
+            [found_number_to_jobj(SavedNumber) | Acc];
+        {'error', _R} ->
+            lager:debug("skipping number ~s: ~p", [knm_phone_number:number(PhoneNumber), _R]),
+            Acc
     end.
 
--spec found_numbers_to_jobjs(knm_number:knm_numbers()) -> kz_json:objects().
-found_numbers_to_jobjs(Numbers) ->
-    [found_number_to_jobj(Number) || Number <- Numbers].
-
 -spec found_number_to_jobj(knm_number:knm_number()) -> kz_json:object().
 found_number_to_jobj(Number) ->
     PhoneNumber = knm_number:phone_number(Number),
-    found_number_to_jobj(PhoneNumber, knm_phone_number:module_name(PhoneNumber)).
-
--spec found_number_to_jobj(knm_phone_number:knm_phone_number(), ne_binary()) ->
-                                  kz_json:object().
-found_number_to_jobj(PhoneNumber, ?CARRIER_MANAGED) ->
     CarrierData = knm_phone_number:carrier_data(PhoneNumber),
-    kz_json:from_list(
-      props:filter_undefined(
-              [{<<"number">>, knm_phone_number:number(PhoneNumber)}
-               ,{<<"rate">>, kz_json:get_value(<<"rate">>, CarrierData, <<"1">>)}
-               ,{<<"activation_charge">>, kz_json:get_value(<<"activation_charge">>, CarrierData, <<"0">>)}
-              ])
-     );
-found_number_to_jobj(PhoneNumber, _Carrier) ->
     DID = knm_phone_number:number(PhoneNumber),
-    CarrierData = knm_phone_number:carrier_data(PhoneNumber),
-    AssignTo = knm_phone_number:assign_to(PhoneNumber),
-
-    kz_json:set_values(
-      props:filter_undefined(
-        [{<<"number">>, DID}
-         ,{<<"activation_charge">>, activation_charge(DID, AssignTo)}
-        ]
-       )
-      ,CarrierData
-     ).
+    case knm_phone_number:module_name(PhoneNumber) of
+        ?CARRIER_MANAGED ->
+            kz_json:from_list(
+              props:filter_undefined(
+                [{<<"number">>, DID}
+                ,{<<"rate">>, kz_json:get_value(<<"rate">>, CarrierData, <<"1">>)}
+                ,{<<"activation_charge">>, kz_json:get_value(<<"activation_charge">>, CarrierData, <<"0">>)}
+                ,{<<"state">>, knm_phone_number:state(PhoneNumber)}
+                ])
+             );
+        _Carrier ->
+            AssignTo = knm_phone_number:assign_to(PhoneNumber),
+            Add = props:filter_undefined(
+                    [{<<"number">>, DID}
+                    ,{<<"activation_charge">>, activation_charge(DID, AssignTo)}
+                    ,{<<"state">>, knm_phone_number:state(PhoneNumber)}
+                    ]),
+            kz_json:set_values(Add, CarrierData)
+    end.
```

Comment:
I'm not sure why OTHER and MANAGED are treated differently. That's @jamesaimonetti's code from a year ago.